<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8" />
<title>DefaultExceptionDescriptorTest xref</title>
<link type="text/css" rel="stylesheet" href="../../../../../../stylesheet.css" />
</head>
<body>
<div id="overview"><a href="../../../../../../../testapidocs/com/github/triceo/splitlog/splitters/exceptions/DefaultExceptionDescriptorTest.html">View Javadoc</a></div><pre>

<a class="jxr_linenumber" name="1" href="#1">1</a>   <strong class="jxr_keyword">package</strong> com.github.triceo.splitlog.splitters.exceptions;
<a class="jxr_linenumber" name="2" href="#2">2</a>   
<a class="jxr_linenumber" name="3" href="#3">3</a>   <strong class="jxr_keyword">import</strong> java.net.ConnectException;
<a class="jxr_linenumber" name="4" href="#4">4</a>   
<a class="jxr_linenumber" name="5" href="#5">5</a>   <strong class="jxr_keyword">import</strong> org.assertj.core.api.Assertions;
<a class="jxr_linenumber" name="6" href="#6">6</a>   <strong class="jxr_keyword">import</strong> org.junit.Test;
<a class="jxr_linenumber" name="7" href="#7">7</a>   
<a class="jxr_linenumber" name="8" href="#8">8</a>   <strong class="jxr_keyword">import</strong> com.github.triceo.splitlog.api.ExceptionDescriptor;
<a class="jxr_linenumber" name="9" href="#9">9</a>   
<a class="jxr_linenumber" name="10" href="#10">10</a>  <strong class="jxr_keyword">public</strong> <strong class="jxr_keyword">class</strong> <a href="../../../../../../com/github/triceo/splitlog/splitters/exceptions/DefaultExceptionDescriptorTest.html">DefaultExceptionDescriptorTest</a> {
<a class="jxr_linenumber" name="11" href="#11">11</a>  
<a class="jxr_linenumber" name="12" href="#12">12</a>      <strong class="jxr_keyword">private</strong> <strong class="jxr_keyword">void</strong> assertException(<strong class="jxr_keyword">final</strong> ExceptionDescriptor ex, <strong class="jxr_keyword">final</strong> String expectedClassName,
<a class="jxr_linenumber" name="13" href="#13">13</a>          <strong class="jxr_keyword">final</strong> String expectedMessage) {
<a class="jxr_linenumber" name="14" href="#14">14</a>          Assertions.assertThat(ex).isNotNull();
<a class="jxr_linenumber" name="15" href="#15">15</a>          Assertions.assertThat(ex.getExceptionClassName()).isEqualTo(expectedClassName);
<a class="jxr_linenumber" name="16" href="#16">16</a>          Assertions.assertThat(ex.getMessage()).isEqualTo(expectedMessage);
<a class="jxr_linenumber" name="17" href="#17">17</a>      }
<a class="jxr_linenumber" name="18" href="#18">18</a>  
<a class="jxr_linenumber" name="19" href="#19">19</a>      @Test
<a class="jxr_linenumber" name="20" href="#20">20</a>      <strong class="jxr_keyword">public</strong> <strong class="jxr_keyword">void</strong> testMaven() {
<a class="jxr_linenumber" name="21" href="#21">21</a>          <strong class="jxr_keyword">final</strong> ExceptionDescriptor ex = DefaultExceptionDescriptor.parseStackTrace((ExceptionParserTest
<a class="jxr_linenumber" name="22" href="#22">22</a>                  .parseIntoLines(<strong class="jxr_keyword">this</strong>.getClass().getResourceAsStream(<span class="jxr_string">"exception-maven.txt"</span>))));
<a class="jxr_linenumber" name="23" href="#23">23</a>          <strong class="jxr_keyword">this</strong>.assertException(
<a class="jxr_linenumber" name="24" href="#24">24</a>                  ex,
<a class="jxr_linenumber" name="25" href="#25">25</a>                  <span class="jxr_string">"java.lang.IllegalStateException"</span>,
<a class="jxr_linenumber" name="26" href="#26">26</a>                  <span class="jxr_string">"There is no context available for qualifier org.jboss.arquillian.drone.api.annotation.Default. Available contexts are []."</span>);
<a class="jxr_linenumber" name="27" href="#27">27</a>          Assertions.assertThat(ex.getExceptionClass()).isEqualTo(IllegalStateException.<strong class="jxr_keyword">class</strong>);
<a class="jxr_linenumber" name="28" href="#28">28</a>          Assertions.assertThat(ex.getCause()).isNull();
<a class="jxr_linenumber" name="29" href="#29">29</a>      }
<a class="jxr_linenumber" name="30" href="#30">30</a>  
<a class="jxr_linenumber" name="31" href="#31">31</a>      @Test
<a class="jxr_linenumber" name="32" href="#32">32</a>      <strong class="jxr_keyword">public</strong> <strong class="jxr_keyword">void</strong> testWithCauses() {
<a class="jxr_linenumber" name="33" href="#33">33</a>          <strong class="jxr_keyword">final</strong> ExceptionDescriptor ex = DefaultExceptionDescriptor.parseStackTrace((ExceptionParserTest
<a class="jxr_linenumber" name="34" href="#34">34</a>                  .parseIntoLines(<strong class="jxr_keyword">this</strong>.getClass().getResourceAsStream(<span class="jxr_string">"exception-with-causes.txt"</span>))));
<a class="jxr_linenumber" name="35" href="#35">35</a>          <strong class="jxr_keyword">this</strong>.assertException(ex, <span class="jxr_string">"org.jboss.qa.bpms.rest.wb.RequestFailureException"</span>,
<a class="jxr_linenumber" name="36" href="#36">36</a>                  <span class="jxr_string">"java.net.ConnectException: Connection refused"</span>);
<a class="jxr_linenumber" name="37" href="#37">37</a>          Assertions.assertThat(ex.getExceptionClass()).isNull(); <em class="jxr_comment">// highly likely</em>
<a class="jxr_linenumber" name="38" href="#38">38</a>          <strong class="jxr_keyword">final</strong> ExceptionDescriptor cause = ex.getCause();
<a class="jxr_linenumber" name="39" href="#39">39</a>          <strong class="jxr_keyword">this</strong>.assertException(cause, <span class="jxr_string">"java.net.ConnectException"</span>, <span class="jxr_string">"Connection refused"</span>);
<a class="jxr_linenumber" name="40" href="#40">40</a>          Assertions.assertThat(cause.getExceptionClass()).isEqualTo(ConnectException.<strong class="jxr_keyword">class</strong>);
<a class="jxr_linenumber" name="41" href="#41">41</a>          Assertions.assertThat(cause.getCause()).isNull();
<a class="jxr_linenumber" name="42" href="#42">42</a>  
<a class="jxr_linenumber" name="43" href="#43">43</a>      }
<a class="jxr_linenumber" name="44" href="#44">44</a>  
<a class="jxr_linenumber" name="45" href="#45">45</a>      @Test
<a class="jxr_linenumber" name="46" href="#46">46</a>      <strong class="jxr_keyword">public</strong> <strong class="jxr_keyword">void</strong> testWithCauses2() {
<a class="jxr_linenumber" name="47" href="#47">47</a>          <strong class="jxr_keyword">final</strong> ExceptionDescriptor ex = DefaultExceptionDescriptor.parseStackTrace((ExceptionParserTest
<a class="jxr_linenumber" name="48" href="#48">48</a>                  .parseIntoLines(<strong class="jxr_keyword">this</strong>.getClass().getResourceAsStream(<span class="jxr_string">"exception-with-causes2.txt"</span>))));
<a class="jxr_linenumber" name="49" href="#49">49</a>          <strong class="jxr_keyword">this</strong>.assertException(ex, <span class="jxr_string">"javax.servlet.ServletException"</span>, <span class="jxr_string">"Something bad happened"</span>);
<a class="jxr_linenumber" name="50" href="#50">50</a>          Assertions.assertThat(ex.getCause()).isNotNull();
<a class="jxr_linenumber" name="51" href="#51">51</a>          <em class="jxr_comment">// TODO could use a bit more validation; but at least it parses</em>
<a class="jxr_linenumber" name="52" href="#52">52</a>      }
<a class="jxr_linenumber" name="53" href="#53">53</a>  
<a class="jxr_linenumber" name="54" href="#54">54</a>      @Test
<a class="jxr_linenumber" name="55" href="#55">55</a>      <strong class="jxr_keyword">public</strong> <strong class="jxr_keyword">void</strong> testWithLinePrefix() {
<a class="jxr_linenumber" name="56" href="#56">56</a>          <strong class="jxr_keyword">final</strong> ExceptionDescriptor ex = DefaultExceptionDescriptor.parseStackTrace((ExceptionParserTest
<a class="jxr_linenumber" name="57" href="#57">57</a>                  .parseIntoLines(<strong class="jxr_keyword">this</strong>.getClass().getResourceAsStream(<span class="jxr_string">"exception-with-line-prefix.txt"</span>))));
<a class="jxr_linenumber" name="58" href="#58">58</a>          <strong class="jxr_keyword">this</strong>.assertException(
<a class="jxr_linenumber" name="59" href="#59">59</a>                  ex,
<a class="jxr_linenumber" name="60" href="#60">60</a>                  <span class="jxr_string">"org.switchyard.SwitchYardException"</span>,
<a class="jxr_linenumber" name="61" href="#61">61</a>                  <span class="jxr_string">"SWITCHYARD014032: Operation fail does not exist for service {urn:ledegen:operation-selector-service:1.0}SimpleHttpGreetingGateway"</span>);
<a class="jxr_linenumber" name="62" href="#62">62</a>          Assertions.assertThat(ex.getCause()).isNull();
<a class="jxr_linenumber" name="63" href="#63">63</a>          <em class="jxr_comment">// TODO could use a bit more validation; but at least it parses</em>
<a class="jxr_linenumber" name="64" href="#64">64</a>      }
<a class="jxr_linenumber" name="65" href="#65">65</a>  }
</pre>
<hr/><div id="footer">This page was automatically generated by <a href="http://maven.apache.org/">Maven</a></div></body>
</html>

